### ISSUE TITLE
[BUG/ISSUE] Benign error in MAPL_Cap at end of run

### ISSUE THREAD

**Issue by kilicomu** [geoschem/GCHP#88]
### Description of the problem
At the end of my GCHP runs, after the ExtData timer output, I am seeing `MAPL_Cap` errors. e.g for a 30 core, single node run:
```
  Times for EXTDATA
TOTAL                   :     245.290     245.797     247.254
Initialize              :      39.519      39.526      39.544
Run                     :     205.751     206.261     207.722
-Read_Loop              :     205.136     205.507     207.032
--CheckUpd              :       0.558       0.604       0.642
--Read                  :       0.000       0.000       0.000
--GridCreate            :       0.000       0.000       0.000
--IclientWait           :       0.000       0.000       0.000
--PRead                 :     194.958     195.245     196.701
---CreateCFIO           :       0.002       0.006       0.010
---prefetch             :      28.186      28.201      28.213
----add-collection      :       0.000       0.000       0.000
----make-reference      :       0.000       0.000       0.000
----RegridStore         :       0.000       0.000       0.000
----request             :       0.000       0.000       0.000
---IclientDone          :      81.141      82.381      83.847
----RegridApply         :       0.000       0.000       0.000
---read-prefetch        :      82.968      84.657      86.171
--Swap                  :       0.022       0.029       0.040
--Bracket               :       9.161       9.190       9.215
-Interpolate            :       0.612       0.750       0.810
GenInitTot              :       0.005       0.006       0.007
--GenInitMine           :       0.005       0.006       0.006
GenRunTot               :     205.761     206.271     207.733
--GenRunMine            :     205.761     206.270     207.732
GenFinalTot             :       0.004       0.005       0.005
--GenFinalMine          :       0.004       0.005       0.005
GenRecordTot            :       0.000       0.000       0.000
--GenRecordMine         :       0.000       0.000       0.000
GenRefreshTot           :       0.000       0.000       0.000
--GenRefreshMine        :       0.000       0.000       0.000
  
 NOT using buffer I/O for file: cap_restart
pe=00023 FAIL at line=00157    MAPL_Cap.F90                             <status=126563264>
pe=00023 FAIL at line=00131    MAPL_Cap.F90                             <status=126563264>
pe=00023 FAIL at line=00029    GCHPctm.F90                              <status=126563264>
pe=00025 FAIL at line=00157    MAPL_Cap.F90                             <status=109577088>
pe=00025 FAIL at line=00131    MAPL_Cap.F90                             <status=109577088>
pe=00025 FAIL at line=00029    GCHPctm.F90                              <status=109577088>
pe=00026 FAIL at line=00157    MAPL_Cap.F90                             <status=108063184>
pe=00026 FAIL at line=00131    MAPL_Cap.F90                             <status=108063184>
pe=00026 FAIL at line=00029    GCHPctm.F90                              <status=108063184>
pe=00027 FAIL at line=00157    MAPL_Cap.F90                             <status=112244688>
pe=00027 FAIL at line=00131    MAPL_Cap.F90                             <status=112244688>
pe=00027 FAIL at line=00029    GCHPctm.F90                              <status=112244688>
pe=00005 FAIL at line=00157    MAPL_Cap.F90                             <status=114534336>
pe=00007 FAIL at line=00157    MAPL_Cap.F90                             <status=109956704>
pe=00007 FAIL at line=00131    MAPL_Cap.F90                             <status=109956704>
pe=00007 FAIL at line=00029    GCHPctm.F90                              <status=109956704>
pe=00010 FAIL at line=00157    MAPL_Cap.F90                             <status=109966176>
pe=00010 FAIL at line=00131    MAPL_Cap.F90                             <status=109966176>
pe=00010 FAIL at line=00029    GCHPctm.F90                              <status=109966176>
pe=00013 FAIL at line=00157    MAPL_Cap.F90                             <status=90365936>
pe=00013 FAIL at line=00131    MAPL_Cap.F90                             <status=90365936>
pe=00013 FAIL at line=00029    GCHPctm.F90                              <status=90365936>
pe=00015 FAIL at line=00157    MAPL_Cap.F90                             <status=120395056>
pe=00015 FAIL at line=00131    MAPL_Cap.F90                             <status=120395056>
pe=00015 FAIL at line=00029    GCHPctm.F90                              <status=120395056>
pe=00017 FAIL at line=00157    MAPL_Cap.F90                             <status=105551200>
pe=00017 FAIL at line=00131    MAPL_Cap.F90                             <status=105551200>
pe=00017 FAIL at line=00029    GCHPctm.F90                              <status=105551200>
pe=00021 FAIL at line=00157    MAPL_Cap.F90                             <status=107180896>
pe=00021 FAIL at line=00131    MAPL_Cap.F90                             <status=107180896>
pe=00021 FAIL at line=00029    GCHPctm.F90                              <status=107180896>
pe=00028 FAIL at line=00157    MAPL_Cap.F90                             <status=116531024>
pe=00028 FAIL at line=00131    MAPL_Cap.F90                             <status=116531024>
pe=00028 FAIL at line=00029    GCHPctm.F90                              <status=116531024>
pe=00029 FAIL at line=00157    MAPL_Cap.F90                             <status=106845472>
pe=00029 FAIL at line=00131    MAPL_Cap.F90                             <status=106845472>
pe=00029 FAIL at line=00029    GCHPctm.F90                              <status=106845472>
pe=00001 FAIL at line=00157    MAPL_Cap.F90                             <status=124487440>
pe=00001 FAIL at line=00131    MAPL_Cap.F90                             <status=124487440>
pe=00001 FAIL at line=00029    GCHPctm.F90                              <status=124487440>
pe=00002 FAIL at line=00157    MAPL_Cap.F90                             <status=123311376>
pe=00002 FAIL at line=00131    MAPL_Cap.F90                             <status=123311376>
pe=00002 FAIL at line=00029    GCHPctm.F90                              <status=123311376>
pe=00003 FAIL at line=00157    MAPL_Cap.F90                             <status=105751040>
pe=00003 FAIL at line=00131    MAPL_Cap.F90                             <status=105751040>
pe=00003 FAIL at line=00029    GCHPctm.F90                              <status=105751040>
pe=00004 FAIL at line=00157    MAPL_Cap.F90                             <status=98386912>
pe=00004 FAIL at line=00131    MAPL_Cap.F90                             <status=98386912>
pe=00004 FAIL at line=00029    GCHPctm.F90                              <status=98386912>
pe=00006 FAIL at line=00157    MAPL_Cap.F90                             <status=117969776>
pe=00006 FAIL at line=00131    MAPL_Cap.F90                             <status=117969776>
pe=00006 FAIL at line=00029    GCHPctm.F90                              <status=117969776>
pe=00008 FAIL at line=00157    MAPL_Cap.F90                             <status=115786544>
pe=00008 FAIL at line=00131    MAPL_Cap.F90                             <status=115786544>
pe=00008 FAIL at line=00029    GCHPctm.F90                              <status=115786544>
pe=00009 FAIL at line=00157    MAPL_Cap.F90                             <status=130416032>
pe=00009 FAIL at line=00131    MAPL_Cap.F90                             <status=130416032>
pe=00009 FAIL at line=00029    GCHPctm.F90                              <status=130416032>
pe=00011 FAIL at line=00157    MAPL_Cap.F90                             <status=121717920>
pe=00011 FAIL at line=00131    MAPL_Cap.F90                             <status=121717920>
pe=00011 FAIL at line=00029    GCHPctm.F90                              <status=121717920>
pe=00012 FAIL at line=00157    MAPL_Cap.F90                             <status=118612512>
pe=00012 FAIL at line=00131    MAPL_Cap.F90                             <status=118612512>
pe=00012 FAIL at line=00029    GCHPctm.F90                              <status=118612512>
pe=00014 FAIL at line=00157    MAPL_Cap.F90                             <status=99530368>
pe=00014 FAIL at line=00131    MAPL_Cap.F90                             <status=99530368>
pe=00014 FAIL at line=00029    GCHPctm.F90                              <status=99530368>
pe=00016 FAIL at line=00157    MAPL_Cap.F90                             <status=101856992>
pe=00016 FAIL at line=00131    MAPL_Cap.F90                             <status=101856992>
pe=00016 FAIL at line=00029    GCHPctm.F90                              <status=101856992>
pe=00018 FAIL at line=00157    MAPL_Cap.F90                             <status=114836960>
pe=00018 FAIL at line=00131    MAPL_Cap.F90                             <status=114836960>
pe=00018 FAIL at line=00029    GCHPctm.F90                              <status=114836960>
pe=00019 FAIL at line=00157    MAPL_Cap.F90                             <status=117949808>
pe=00019 FAIL at line=00131    MAPL_Cap.F90                             <status=117949808>
pe=00019 FAIL at line=00029    GCHPctm.F90                              <status=117949808>
pe=00020 FAIL at line=00157    MAPL_Cap.F90                             <status=110433600>
pe=00020 FAIL at line=00131    MAPL_Cap.F90                             <status=110433600>
pe=00020 FAIL at line=00029    GCHPctm.F90                              <status=110433600>
pe=00022 FAIL at line=00157    MAPL_Cap.F90                             <status=113983248>
pe=00022 FAIL at line=00131    MAPL_Cap.F90                             <status=113983248>
pe=00022 FAIL at line=00029    GCHPctm.F90                              <status=113983248>
pe=00024 FAIL at line=00157    MAPL_Cap.F90                             <status=99507984>
pe=00024 FAIL at line=00131    MAPL_Cap.F90                             <status=99507984>
pe=00024 FAIL at line=00029    GCHPctm.F90                              <status=99507984>
pe=00005 FAIL at line=00131    MAPL_Cap.F90                             <status=114534336>
pe=00005 FAIL at line=00029    GCHPctm.F90                              <status=114534336>
Model Throughput:       26.027 days per day
pe=00000 FAIL at line=00157    MAPL_Cap.F90                             <status=118968240>
pe=00000 FAIL at line=00131    MAPL_Cap.F90                             <status=118968240>
pe=00000 FAIL at line=00029    GCHPctm.F90                              <status=118968240>
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 17 in communicator MPI_COMM_WORLD
with errorcode 0.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
[node103.pri.viking.alces.network:131768] 29 more processes have sent help message help-mpi-api.txt / mpi-abort
[node103.pri.viking.alces.network:131768] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
```

The run outputs appear to be produced just fine! The error is apparently coming from `run_ensemble` in `MAPL_Cap.F90` - I'm not familiar with MAPL so I'll have to do some digging to figure out any more about the source of the error.

Any suggestions appreciated!

### GEOS-Chem version
GCHP `dev/13.0.0-final` (577b2d82)

### Description of code modifications
None

### Log files
- Build log (if applicable): [cmake_make_output.log](https://github.com/geoschem/GCHP/files/6075453/cmake_make_output.log)
- Run log (if applicable): [gchp_c48_1day_30cores.log](https://github.com/geoschem/GCHP/files/6076082/gchp_c48_1day_30cores.log)
- Run script (if applicable): [gchp.sbatch.txt](https://github.com/geoschem/GCHP/files/6076078/gchp.sbatch.txt) & [runConfig.sh.txt](https://github.com/geoschem/GCHP/files/6076283/runConfig.sh.txt)

### Software versions
- Compilers (Intel or GNU, and version): GNU v9.3.0
- NetCDF version: NetCDF-C 4.7.4, NetCDF-Fortran 4.5.2
- MPI version: OpenMPI 4.0.3
- ESMF version: ESMF 8.0.1

**Comment by LiamBindle** [comment]
Thanks for opening this @kilicomu! I've ran into this myself before too, but I haven't had a chance to dig into it. It can be an issue for timing tests, because the error throws before the timing stats print. Besides that though, I've found it's just a nuisance. 

As you mentioned, here's a quick link to the erroring line: https://github.com/geoschem/MAPL/blob/a9590432be58889e4277253c7872d982980e4bb3/base/MAPL_Cap.F90#L157

I'm not familiar with `run_ensemble` either. @lizziel Do you know what the issue might be?

**Comment by LiamBindle** [comment]
I just updated the title&mdash;feel free to update it if you'd like.

**Comment by kilicomu** [comment]
For additional info I ran the same run with the MAPL debug flags on (man, that log is *large*): [gchp_c48_1day_30cores_MAPL_debug.log.zip](https://github.com/geoschem/GCHP/files/6076878/gchp_c48_1day_30cores_MAPL_debug.log.zip)

**Comment by lizziel** [comment]
This doesn't ring a bell but I'll bring it up with MAPL developers later today. They might have some insight into the cause, and maybe a fix.

**Comment by lizziel** [comment]
@mathomp4, have you seen this error in `MAPL_Cap.F90` before? GCHP gets through the run normally otherwise but prints errors from lines [157](https://github.com/geoschem/MAPL/blob/a9590432be58889e4277253c7872d982980e4bb3/base/MAPL_Cap.F90#L157) and [131](https://github.com/geoschem/MAPL/blob/a9590432be58889e4277253c7872d982980e4bb3/base/MAPL_Cap.F90#L131).

**Comment by mathomp4** [comment]
@lizziel Huh. Nope. This is new to me. Time to ping @bena-nasa and @tclune I think (and you might want to file an issue on the MAPL repo...)

**Comment by lizziel** [comment]
Following discussion, it looks like this is due to not setting `rc=status` in the `run_member` subroutine. It is also needed in `run_ensemble` and `finalize_mpi` subroutines. MAPL developers are going to clean up the file in the next version and thank you for bringing this to their attention.

**Comment by lizziel** [comment]
The MAPL update to fix this is in MAPL v2.6.4 which will be included in GCHP 13.1.0.
